### PR TITLE
1ES pipeline migration

### PR DESCRIPTION
**Pull request by melytc**
- Added the MicroBuildTemplate repo
- Extended the template
- Added the API scan and Code sign

here is the pipeline with changes: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=9304525&view=results

### CODE REVIEW

File: eng/official.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 resources:
   repositories:
+  - repository: MicroBuildTemplate
```
Comment (smitpatel) [inline eng/official.yml:5]: Also include link to the pipeline as comment on top of the file.

File: eng/official.yml [hunk 2]
```diff
@@ -13,51 +17,117 @@ variables:
...
+  template: azure-pipelines/MicroBuild.1ES.Official.yml@MicroBuildTemplate
+  parameters:
+    sdl:
+      sourceanalysisPool: VSEngSS-MicroBuild2022-1ES
+      tsa:
+        enabled: true
+        configFile: $(Build.SourcesDirectory)/eng/pipelines/configuration/TSAConfig.gdntsa
+      binskim:
+        enabled: true
+        analyzeTargetGlob: $(Build.SourcesDirectory)/artifacts/$(BuildConfiguration)/bin/Dlls/**.dll
```
Comment (smitpatel) [inline eng/official.yml:35]: This is not true for arcade based repos

File: eng/official.yml [hunk 3]
```diff
@@ -13,51 +17,117 @@ variables:
...
+        enabled: true
+      policheck:
+        enabled: true
+    stages:
+    - stage: AdditionalStages
+      displayName: Build
+      jobs:
+      - job: AdditionalSteps
+        steps:
+          - checkout: self
```
Comment (smitpatel) [inline eng/official.yml:46]: You dont need checkout in yaml based pipeline

File: eng/official.yml [hunk 4]
```diff
@@ -13,51 +17,117 @@ variables:
...
+            clean: true
+          - task: MicroBuildSigningPlugin@4
```
Comment (smitpatel) [inline eng/official.yml:48]: remove these 2 plugin installations. template should do it for you.

**Review by haileymck — APPROVED** [review]

**Pull request merged**